### DISCUSSION TITLE
Questions about EP detetor for LLR calculation

### DISCUSSION THREAD

**Discussion by HaoChang1** [NVlabs/sionna#526] [Q&A] [answered]
Hi,

When I try to use Sionna for MIMO detection in coding system. I noticed that the Logits/LLR calculation in EP restricts the minimum variance to 10^-6. I there any particular reason for that?
Is that because samll variance in LLR calculation resulting in Large LLR values that will impact the decoding process?

**Comment by merlinND** [comment]
Hello @HaoChang1,

Could you please share how your question was resolved so that future people encountering this question also benefit from your finding?

**Reply by HaoChang1** [reply]
Hi,

This problem raises due to my misunderstanding. The actual mean and variance used for LLR calculation is obtained in the EP observation module. However, the limitation of variance to 10^-6 is in the EP estimation module.

I come accross another question. Why the outperformance of EP over LMMSE in the coded MIMO system degrade as compared to in the uncoded one ? (i.e., EP outperforms LMMSE over 10dB for uncoded BER but only outperforms 3-5dB in coded BER).
I am wondering is this becasue of LMMSE symbol estimates can return x_hat = x +n module for LLR calculation but EP can not?

**Reply by jhoydis** [reply, marked as answer]
Hi @HaoChang1,
 
There might be two reasons for what you observe:
 
1) The quality of LLRs of EP might not be as good as those of LMMSE
 
2) This depends at which SNR the waterfall region of the code starts, in other words the chosen code rate. In most cases, this corresponds to a rather low SNR at which the performance difference between LMMSE and EP is not so large. That is the reason why many papers looking at uncoded BER performance overestimate the performance gains of their MIMO detection scheme.

**Reply by HaoChang1** [reply]
Thanks ! appreciate your answer.
